msal/src/main/java/com/microsoft/identity/client/PublicClientApplication.java

```diff
@@ -784,7 +785,13 @@ public void acquireTokenSilentAsync(@NonNull final String[] scopes,
         String requestAuthority = authority;
 
         if (StringUtil.isEmpty(requestAuthority)) {
-            requestAuthority = Authority.getAuthorityFromAccount(account);
+            if (!StringUtil.isEmpty(mAuthorityString)
+                    && Authority.getAuthorityFromAuthorityUrl(mAuthorityString) instanceof AzureActiveDirectoryB2CAuthority) {
```

Comment:
Authority.getAuthorityFromAuthorityUrl can throw an illegalargumentexception, since we are doing this last moment, see if you can make this more defensive

Reply:
also can you add a comment on why this logic is this way